### DISCUSSION TITLE
How do I download our Edgedb schema as a TS file for local development? Because I don't want others to be unable to modify my schema.

### DISCUSSION THREAD

**Discussion by Tk8and15** [geldata/gel#4879] [Q&A]
How do I download our Edgedb schema as a TS file for local development? I want to use the builder query form for project development.

**Comment by raddevon** [comment]
I think I understand you have a remote EdgeDB instance that you want to run the generator on to generate a query builder for your TS project. If that's correct and if you have the CLI linked to your remote instance, you could run the generator with `--instance <remote-instance-name>` to generate from that instance.

**Reply by Tk8and15** [reply]
I have a remote instance, which has been linked. What I want is edgedb as a database, and then typescript as a client. I want to separate them and call my edgedb through typescript client. I don't want anyone to modify my edgedb schame.

**Reply by raddevon** [reply]
Our [TS client connection documentation](https://www.edgedb.com/docs/clients/js/driver#connections) might be what you need. You will need to set [environment variables](https://www.edgedb.com/docs/reference/connection#specifying-an-instance) so that your TS client can connect to the database if it's remote.

**Reply by Tk8and15** [reply]
Can you teach me how to set it up?

**Reply by raddevon** [reply]
Since you have your remote instance linked, you could just specify `EDGEDB_INSTANCE`. If you're on a Unix-based OS, you can run `export EDGEDB_INSTANCE=<remote-instance-name>` before you run your TS to get it to connect to the remote instance.

**Reply by Tk8and15** [reply]
I'm using Windows, this is my configuration on docker, so how do I set my environment variables? My remote port number is 5656.
![Uploading image.png…]()
 Here are my local and remote instances.
![Uploading image.png…]()
这是我的TS客户端配置代码。
![Uploading image.png…]()
Please help me see what's wrong.
The listening URL is 4001.
![Uploading image.png…]()

**Reply by raddevon** [reply]
Under Powershell, you can set an environment variable like this:
```powershell
$Env:EDGEDB_PORT = 5656
```
which would be the setting for your port based on what you've told me. I can't see any of the images you've tried to share here, so I'm not sure what your other settings would be. Just match up the environment variables in that article I linked with your connection details, setting each one, and you should be good to go!

**Reply by Tk8and15** [reply]
I'm using Windows, this is my configuration on docker, so how do I set my environment variables? My remote port number is 5656.
![image](https://user-images.githubusercontent.com/118907993/211696381-0a551c1c-e84e-4e47-b44c-bdca8a4b7ace.png)
Here are my local and remote instances.
![image](https://user-images.githubusercontent.com/118907993/211696417-f8f7e2d9-1630-4ffe-bec9-9becdcc4575b.png)
TS client.
![image](https://user-images.githubusercontent.com/118907993/211696488-45d5f296-e3ac-4ca1-b0e6-f0b87ee579ae.png)
The listening URL is 4001.
![image](https://user-images.githubusercontent.com/118907993/211696521-f25e2f81-ce62-434e-bf97-56c870d1446a.png)

**Reply by raddevon** [reply]
Try this. Run `edgedb instance credentials -I <instance-name> --insecure-dsn` (filling in your instance's name). Take the output of that command and use it to set the DSN environment variable like this: `$Env:EDGEDB_DSN = <dsn>` (replacing `<dsn>` with what you got from the previous command). Then run `$Env:EDGEDB_CLIENT_TLS_SECURITY = 'insecure'`.
